brillig_bytecode/src/opcodes.rs

```diff
@@ -117,6 +117,12 @@ pub struct OracleData {
     pub output_values: Vec<FieldElement>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct OracleInput {
+    pub register_mem_index: RegisterMemIndex,
+    pub length: usize,
+}
```

Comment:
I'm not a fan of this representation to handle registers and arrays. A register value is not a 0-length array, it is a fundamentally different type so we should represent it as such. This should be an enum that is either Register or Array. This representation would allow easier extension in the future and prevent future bugs from forgetting to check the length field.